bridgetown-website/src/_docs/plugins.md

```diff
@@ -210,29 +202,84 @@ module MySite
 end
 ```
 
+## Zeitwerk and Autoloading
+
+Bridgetown 1.0 brings with it a new autoloading mechanism using [Zeitwerk](https://github.com/fxn/zeitwerk), the same code loader used by Rails and many other Ruby-based projects. Zeitwerk uses a specific naming convension so the paths of your Ruby files and the namespaces/modules/classes of your Ruby code are aligned. For example:
+
+```
+plugins/my_plugin.rb         -> MyPlugin
+plugins/my_plugin/foo.rb     -> MyPlugin::Foo
+plugins/my_plugin/bar_baz.rb -> MyPlugin::BarBaz
+plugins/my_plugin/woo/zoo.rb -> MyPlugin::Woo::Zoo
+```
+
+You can read more about [Zeitwerk's file conventions here](https://github.com/fxn/zeitwerk#file-structure).
+
+{% rendercontent "docs/note", title: "Take Me Back" %}
+  If you run into any problems with Zeitwerk after upgrading your Bridgetown project from pre-1.0, you can switch to the previous plugin loading method by adding `plugins_use_zeitwerk: false` to your `bridgetown.config.yml`. Or you can try using the `loader_collapsed_paths` setting as described below.
+{% endrendercontent %}
+
+In addition to the `plugins` folder provided by default, **you can add your own folders** with autoloading support! Simply add to the `autoload_paths` setting in your config YAML:
+
+```yaml
+autoload_paths:
+  - loadme
+```
+
+Now any Ruby file in your project's `./loadme` folder will be autoloaded. By default, files in your custom folders not "eager loaded", meaning that the Ruby code isn't actually processed unless/until you access the class or module name of the file somewhere in your code elsewhere. This can improve performance in certain cases. However, if you need to rely on the fact that your Ruby code is always loaded when the site is instantiated, simply set `eager` to true in your config:
+
+```yaml
+autoload_paths:
+  - path: loadme
+    eager: true
+```
+
+There may be times when you want to bypass Zeitwerk's default folder-based namespacing. For example, if you wanted something like this:
+
+```
+plugins/builders/tags.rb   -> Builders::Tags
+plugins/helpers/hashify.rb -> Hashify
+```
+
+where the files in `builders` use a `Builders` namespace, but the files in `helpers` don't use a `Helpers` namespace, you can use the `loader_collapsed_paths` setting:
+
+```yaml
+loader_collapsed_paths:
+  - plugins/helpers
+```
+
+And if you don't want namespacing for _any_ subfolders, you can use a glob pattern:
+
+```yaml
+loader_collapsed_paths:
+  - top_level/*
+```
+
+Thus no files directly in `top_level` as well as any of its subfolders will be namespaced (that is, no `TopLevel` module will be implied).
```

Comment:
Might be worth adding the word "direct" just to make it a bit more clear that deeper subfolders will be namespaced.

```suggestion
Thus no files directly in `top_level` as well as any of its direct subfolders will be namespaced (that is, no `TopLevel` module will be implied).
```